draft-ietf-wimse-client-assertion-in-workload-environments.md

```diff
@@ -148,148 +135,292 @@ capitals, as shown here.
 
 The terms 'workload' and 'container' are used interchangeably.
 
-# Recommendations {#recommendations}
+# Patterns {#patterns}
 
-This specification relies on the use of OAuth 2.0 {{RFC6749}} and
-{{RFC7523}} for client authentication using a JWT.
+## Kubernetes {#kubernetes}
 
-Service Account Tokens used in container orchestration systems are vulnerable to various threats, as outlined below:
+Kubernetes notion of identity is mainly the concept of "service accounts" which are either created explictly or automatically during deployment time. While the mean purpose of these identities is to authenticate access to the Kubernetes Control Plane it is more and more used to authenticate workloads between each other too. The credential format used to authenticate service account is JWT that are signed by the Kubernetes Control Plane.
 
-1. Token theft: Attackers who compromise a workload can steal tokens to impersonate it and gain unauthorized access to resources.
-2. Token reuse: Stolen tokens may be reused within their expiration period to gain repeated unauthorized access. However, the expiration time limits the token reuse time window.
-3. Misconfigured service accounts: mproperly configured service accounts can grant applications excessive privileges.
-4. Theft of token signing key: Attackers gaining control plane access can steal the token signing key, akin to compromising client_id and client_secret in OAuth, potentially accessing all secrets in the orchestration system.
+To programatically use service account tokens workloads can
 
-The following fields are populated in the Service Account Token:
+* use the TokenRequest API of the control plane
 
-1. The 'iss' claim MUST contain a string identifying the worklod orchestrator.
-2. The 'sub' claim MUST contain a string identifying the workload, also serving as the client_id per {{RFC7523}}.
-3. The 'aud' claim MUST identify one or multiple authorization servers intended to receive and authorize the Service Account Token.
+* have the token projected into the file system of the workload. This is commonly referred to as "projected service accout token".
 
-Additional processing requirements are specified in {{RFC7523}}.
+Both options allow workloads to
 
-# Security Considerations
+* specify a custom audience. Possible audiences can be restricted based on control plane policy.
 
-This entire document is about security.
+* specify a custom lifetime. (TODO, is policy possible?)
```

Comment:
https://kubernetes.io/docs/concepts/storage/projected-volumes/#serviceaccounttoken
Yes - one can specify a custom lifetime when projecting the service account token into a pod's filesystem.

Reply:
Thank you!